zoho.go

```diff
@@ -0,0 +1,112 @@
+package zoholab
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/pkg/errors"
+)
+
+const (
+	reportsUri   = "https://analyticsapi.zoho.eu"
+	validJson    = "true"
+	outputFormat = "JSON"
+	errorFormat  = "JSON"
+	apiVersion   = "1.0"
+	addRow       = "ADDROW"
+)
+
+// ZohoService is the struct for the zoho service.
+type ZohoService struct {
+	restyClient *resty.Client
+}
+
+// NewZohoService instantiates a new zoho service.
+func NewZohoService(restyClient *resty.Client) *ZohoService {
```

Comment:
ok hier stellt sich mir immer noch die gleiche frage wie vorher, wieso müssen wir dem consumer der lib auferzwingen resty zu nutzen? wieso wird das nicht innerhalb von dieser function erstellt?

Reply:
Besser Testbar etc. generell wäre ne Erzeugungsabhängigkeit innerhalb von nem ctor auch eher doof. 
Dafür gibts ja Dependency Injection extra. Die Abhängigkeit hat der Consumer sowieso durch das importieren dieser lib.

Reply:
Ich hab den Kommentar vorher noch nicht geschlossen gehabt das ist irgendwie automatisch passiert ich wollte das als nächstes angehen, aber wenn ihr da gegensätzlicher Meinungen seit weis ich auch nicht wirklich was da besser wäre.

Reply:
Die meisten ClientLibs die ich in der Wildnis so sehe zwingen einem das nicht auf, im allerschlimmsten Fall sollte man das mMn wenigstens optional machen, heißt wenn ich will kann ich nen eigenen Client reingeben, ansonsten wird mir Defaultmäßig einer erstellt

Wieso sollte man bei jedem Setup gezwungen sein einen resty client zu erstellen, dem die Middleware dranzuhängen und dann in diese Lib hier rein zu geben? Bei 99% der Fälle wird es genau dieser Ablauf sein, das kann man auch einfach Streamlinen